lib/operations/mongo_client_ops.js

```diff
@@ -600,4 +628,80 @@ function validOptions(options) {
   }
 }
 
+const apmLoggingEvents = {
+  command: ['commandStarted', 'commandSucceeded', 'commandFailed'],
+  sdam: [
+    'serverOpening',
+    'serverDescriptionChanged',
+    'serverHeartbeatStarted',
+    'serverHeartbeatSucceeded',
+    'serverHeartbeatFailed',
+    'serverClosed',
+    'topologyOpening',
+    'topologyClosed',
+    'topologyDescriptionChanged'
+  ]
+};
+
+function apmEventualErrorHandler(e, client) {
+  if (e) {
+    if (!client.topology) throw e;
+    client.close(true, () => {
+      throw e;
+    });
+  }
+}
+
+function getMonitors(monitor) {
+  const allMonitors = ['command', 'sdam'];
+  // enable all monitors if monitor = all
+  let monitors = monitor === 'all' ? allMonitors : monitor;
+  // ensure `monitors` is an array
+  monitors = Array.isArray(monitors) ? monitors : [monitors];
+  return monitors;
+}
+
+function getCustomOut(monitorOut) {
+  let out;
+  if (monitorOut === 'stderr' || monitorOut === 'stdout') {
+    out = process[monitorOut];
+  } else {
+    try {
+      const file = path.resolve(monitorOut);
+      out = createOutFileStream(file);
+    } catch (e) {
+      return e;
+    }
+  }
+
+  return out;
+}
+
+function createOutFileStream(file) {
+  // if file already exists, rename it to avoid overwrite
+  if (fs.existsSync(file)) {
+    const timestamp = new Date().toISOString();
+    fs.renameSync(file, file + '-' + timestamp);
+  }
+
+  return fs.createWriteStream(file);
+}
+
+function attachMonitorListenerType(client, type, stream) {
```

Comment:
`type` here, `monitor` above, this makes it a bit difficult to track. Let's use a single term, and also please document this function